crates/tracing/src/metrics.rs

```diff
@@ -12,15 +12,23 @@ pub fn configure_metrics(
     use wasmcloud_core::OtelProtocol;
 
     let builder: MetricsExporterBuilder = match otel_config.protocol {
-        OtelProtocol::Http => opentelemetry_otlp::new_exporter()
-            .http()
-            .with_protocol(opentelemetry_otlp::Protocol::HttpBinary)
-            .with_endpoint(otel_config.metrics_endpoint())
-            .into(),
-        OtelProtocol::Grpc => opentelemetry_otlp::new_exporter()
-            .tonic()
-            .with_endpoint(otel_config.metrics_endpoint())
-            .into(),
+        OtelProtocol::Http => {
+            let client = crate::get_http_client(otel_config)
+                .context("failed to get an http client for otel metrics exporter")?;
+            opentelemetry_otlp::new_exporter()
+                .http()
+                .with_protocol(opentelemetry_otlp::Protocol::HttpBinary)
+                .with_http_client(client)
+                .with_endpoint(otel_config.metrics_endpoint())
+                .into()
+        }
+        OtelProtocol::Grpc => {
+            // TODO(joonas): Configure tonic::transport::ClientTlsConfig via .with_tls_config(...), passing in additional certificates.
```

Comment:
Does this need to be done as a part of this PR to support additional CA certs for Grpc?

Reply:
I've opted to leave this for a fast follow-up PR that focuses on the tonic story specifically since I'm not aware of any usage of this functionality